codalab/server/bundle_manager.py

```diff
@@ -361,16 +361,8 @@ def _deduct_worker_resources(self, workers_list, running_bundles_info):
         """
         From each worker, subtract resources used by running bundles. Modifies the list.
         """
-
         for worker in workers_list:
             for uuid in worker['run_uuids']:
-                # Verify if the current bundle exists in both the worker table and the bundle table
-                if uuid not in running_bundles_info:
-                    logger.info(
-                        'Bundle {} exists on worker {} but no longer found in the bundle table. '
-                        'Skipping for resource deduction.'.format(uuid, worker['worker_id'])
-                    )
-                    continue
                 bundle_resources = running_bundles_info[uuid]["bundle_resources"]
```

Comment:
@candicegjing can you merge master? I made a change on master to this part where we still do the check and call the db if the bundle is not in the cache. Do you think we need that? Is it possible for a run to be on a worker and not in the cache, but still in the db?

Reply:
@bkgoksel I saw your changes. I think your change is accurate. But the only thing is that it introduces many database queries which will slow down the performance of the scheduler.

Reply:
I agree, I think we want to have both your change (ie cache staged bundles) + query the database if somehow a bundle isn't in that cache either.

This is more of a correctness guarantee, and hopefully we shouldn't hit this case and make that many db queries often (when your changes are added to my change)

Reply:
@bkgoksel The reason that I haven't rebased to master is mainly that I am not sure which size we should pick (accuracy vs performance). I wish there could be a way to fulfill both requirements. I was trying to recall the original reason that leads us to check for the bundle existence during resource deduction. Do you by any chance remember that it's because if the bundle don't exist in the Bundle table, there will be an exception thrown from `compute_bundle_resources` function as it tries to get this bundle from database?

Reply:
@candicegjing for your specific question, I believe that's what was happening. 

As for performance vs accuracy, this is a spot where we can't afford to have any inaccuracy because if we assign too many jobs to a worker, they simply fail. Because of that, any caching/perf improvement has to guarantee not breaking the correctness of resource calculations for workers. 

In this case, I don't think falling back to the db call would affect performance anyways because you cache all the running bundles AND the staged bundles, so it's a very small chance we have a non-cached running bundle. And IF that happens, we  have no choice but to fetch the information for it from the database, as otherwise we'll potentially fail existing runs, which is the worst outcome from a user experience perspective.

Reply:
Yes. I agree. I just rebased to master for now. Let me know if I need to change anything else before merging to master.